Mathlib/RingTheory/SimpleRing/Basic.lean

```diff
@@ -79,6 +80,41 @@ lemma isField_center (A : Type*) [Ring A] [IsSimpleRing A] : IsField (Subring.ce
       _ = y * (a * (x * y)) := by rw [mul_assoc a x y]
       _ = y * a := by rw [hy, mul_one]
 
+lemma injective_ringHom_or_subsingleton_codomain
+    {R S : Type*} [NonAssocRing R] [IsSimpleRing R] [NonAssocRing S]
+    (f : R →+* S) : Function.Injective f ∨ Subsingleton S :=
+  simple.eq_bot_or_eq_top (TwoSidedIdeal.ker f) |>.recOn
```

Comment:
You can probably golf slightly with `Or.imp` instead of `Or.recOn`

Reply:
indeed!